### PR TITLE
[SCAL-86271]- Added reload logic and mixpanel triggers

### PR DESCRIPTION
**Pull request by harsh-ts**
Ticket Id: https://thoughtspot.atlassian.net/browse/SCAL-86271

### CODE REVIEW

File: src/embed/base.ts [hunk 1]
```diff
@@ -453,13 +453,19 @@ export class TsEmbed {
...
         data: any,
     ): typeof TsEmbed.prototype {
-        this.iFrame.contentWindow.postMessage(
-            {
-                type: messageType,
-                data,
-            },
-            this.thoughtSpotHost,
-        );
+        if (messageType === HostEvent.Reload) {
```
Comment (jbhanu-thoughtspot) [inline src/embed/base.ts:456]: Let's move this logic to a different, refer to `src/utils/processData.ts` which is handling the `.on`. Make a similar approach that uses the switch case for different trigger types.

**Review by jbhanu-thoughtspot — CHANGES_REQUESTED** [review]

File: src/embed/base.ts (hunk 1 above)
Comment (harsh-ts) [reply src/embed/base.ts:456]: We are having only 3 cases for trigger event and using switch for only 3 events will be expensive. Let me know your thoughts on this @jbhanu-thoughtspot

File: src/utils/processTrigger.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        case HostEvent.Reload:
+            return reload(iFrame);
+        default:
+            return iFrame.contentWindow.postMessage(
+                {
+                    type: messageType,
+                    data,
+                },
+                thoughtSpotHost,
+            );
```
Comment (jbhanu-thoughtspot) [inline src/utils/processTrigger.ts:29]: Let's move this part as a separate method as well as we might re-use that in other switch cases at a later point

File: src/embed/base.ts (hunk 1 above)
Comment (jbhanu-thoughtspot) [reply src/embed/base.ts]: We are just starting to add the trigger events and pretty soon we will be having many of them and each having a different type of code flow.

File: src/utils/processTrigger.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    thoughtSpotHost: string,
+    data: any,
+) {
+    switch (messageType) {
+        case HostEvent.Reload:
+            return reload(iFrame);
+        default:
+            return postIframeMessage(
+                iFrame,
+                { type: messageType, data },
```
Comment (jbhanu-thoughtspot) [inline src/utils/processTrigger.ts:36]: Pass them as messageType and data as how you are receiving in this method, and form the message object inside the postIframeMessage

**Review by jbhanu-thoughtspot — APPROVED** [review]
lgtm

**Pull request merged**